datadog_checks_base/datadog_checks/base/utils/db/utils.py

```diff
@@ -256,8 +255,19 @@ def __init__(
         self._job_name = job_name
 
     def cancel(self):
+        """
+        Returns only when the async job loop is fully cancelled.
+        future.result() waits until the timeout (10 seconds) to see that the job loop has exited,
+        so we use it here to verify the async job is cancelled before exiting.
+
+        Raises:
+            TimeoutError if the job loop is not cancelled within 10 seconds
+        """
         self._cancel_event.set()
 
+        if self._job_loop_future is not None:
+            self._job_loop_future.result(10)
+
```

Comment:
Hey @jmeunier28 @edengorevoy 👋 Keep in mind that you'll have to bump the minimum base package version in your integrations if your integrations now rely on this change 🙂 (e.g. https://github.com/DataDog/integrations-core/blob/master/mysql/pyproject.toml#L33)